src/main/sensors/battery.c

```diff
@@ -201,10 +218,21 @@ void updateCurrentMeter(int32_t lastUpdateAt, rxConfig_t *rxConfig, uint16_t dea
         case CURRENT_SENSOR_NONE:
             amperage = 0;
             break;
+        case CURRENT_SENSOR_ESC:
+            #ifdef USE_ESC_TELEMETRY
+            amperage = getEscTelemetryCurrent();
+            #endif
+            break;
     }
 
+    #ifdef USE_ESC_TELEMETRY
+    if (batteryConfig->currentMeterType == CURRENT_SENSOR_ESC) {
```

Comment:
Yes, should be checked indeed, when esc telemetry fails this function returns false and has to fallback on onboard sensor.